src/Microsoft.AspNetCore.Routing/EndpointMiddleware.cs

```diff
@@ -32,6 +32,11 @@ public EndpointMiddleware(ILogger<EndpointMiddleware> logger, RequestDelegate ne
         public async Task Invoke(HttpContext httpContext)
         {
             var feature = httpContext.Features.Get<IEndpointFeature>();
+            if (feature == null)
```

Comment:
We wanted to do at startup time? https://github.com/aspnet/Routing/issues/588#issuecomment-403653357 
You seem to have closed that issue for a different change.

Reply:
I didn't see a way to do it at startup. The check in UseMvc that I was thinking about is checking DI registration.

Reply:
Based on @rynowak 's idea, `IApplicationBuilder` exposes `Properties` where we could register a flag if UseDispatcher is called and can verify that property in UseEndpoint.